txnkv/transaction/pipelined_flush.go

```diff
@@ -334,16 +334,34 @@ func (c *twoPhaseCommitter) buildPipelinedResolveHandler(commit bool, resolved *
 		maxBackOff = CommitSecondaryMaxBackoff
 	}
 	regionCache := c.store.GetRegionCache()
+	// the handler function runs in a different goroutine, should copy the required values before it to avoid race.
+	var (
+		startTS           = c.startTS
```

Comment:
It seems they are mostly read-only after initialization. Can they cause data race

Reply:
Actually only `c.txn.diskFullOpt` can cause data race now. But there are no guarantees for those read-only fields, it's a potential risk reading it in another goroutine.

What about init the context here and clone it by `proto.Clone(kvContext).(*kvrpcpb.Context)` in the resolve goroutine?

Reply:
Such guarantees can hardly be achieved in Golang ... I guess we usually ignore them until race is detected.
It's okay we can do this. It just seems a bit strange.